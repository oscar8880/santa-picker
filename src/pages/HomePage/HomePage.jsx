import styles from './HomePage.module.scss';
import React from 'react';
import Text from '../../components/Text/Text';
import Action from '../../components/Action/Action';
import PageWrapper from '../../components/PageWrapper/PageWrapper';
import { Container, Row, Col } from 'react-grid-system';
import presentImage from '../../images/chimneyPresent.png';

const HomePage = () => {
  return (
    <div className={styles.HomePage}>
      <PageWrapper>
        <Container fluid>
          <Row>
            <Col xs={12} md={6} lg={6}>
              <div className={styles.HomePage__StrapLine}>
                <Text appearance="h6" tagName="h1">
                  Secret Santa by email
                </Text>
              </div>
              <ul className={styles.HomePage__BlurbTextList}>
                <li className={styles.HomePage__BlurbTextListItem}>
                  <Text appearance="h5">Guaranteed to be random</Text>
                </li>
                <li className={styles.HomePage__BlurbTextListItem}>
                  <Text appearance="h5">Fully anonymous assignments</Text>
                </li>
                <li className={styles.HomePage__BlurbTextListItem}>
                  <Text appearance="h5">No hat required!</Text>
                </li>
              </ul>
              <div className={styles.HomePage__CTA}>
                <Action tagName="Link" to="/form" appearance="button-secondary">
                  Get Started
                </Action>
              </div>
            </Col>
            <Col xs={12} md={6} lg={5} offset={{ lg: 1 }}>
              <div className={styles.HomePage__Image}>
                <img src={presentImage} alt="Sleigh" width="80%"></img>
              </div>
            </Col>
          </Row>
        </Container>
      </PageWrapper>
    </div>
  );
};

export default HomePage;
