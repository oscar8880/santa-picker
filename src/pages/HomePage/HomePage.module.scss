@import "../../scss/functions";
@import "../../scss/variables";

.HomePage {
  display: flex;
  flex-direction: column;
}
