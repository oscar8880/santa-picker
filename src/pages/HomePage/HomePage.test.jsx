import React from 'react';
import { render } from '@testing-library/react';

import HomePage from './HomePage';

test('matches snapshot', async () => {
  const { asFragment } = render(<HomePage />);

  expect(asFragment()).toMatchSnapshot();
});
