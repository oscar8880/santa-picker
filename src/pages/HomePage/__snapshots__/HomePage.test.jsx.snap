// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="HomePage"
  >
    <div
      class="PageWrapper"
    >
      <div
        style="box-sizing: border-box; position: relative; margin-left: auto; margin-right: auto; padding-left: 15px; padding-right: 15px;"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
          >
            <div
              class="HomePage__StrapLine"
            >
              <h1
                class="Text Text__h6"
              >
                Secret Santa by email
              </h1>
            </div>
            <ul
              class="HomePage__BlurbTextList"
            >
              <li
                class="HomePage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h4"
                >
                  Guaranteed to be random
                </p>
              </li>
              <li
                class="HomePage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h4"
                >
                  Fully anonymous assignments
                </p>
              </li>
              <li
                class="HomePage__BlurbTextListItem"
              >
                <p
                  class="Text Text__h4"
                >
                  No hat required!
                </p>
              </li>
            </ul>
            <div
              class="HomePage__CTA"
            >
              <button
                class="Action Action__Appearance--button-secondary"
              >
                Get Started
              </button>
            </div>
          </div>
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 41.66666666666667%; flex-basis: 41.66666666666667%; flex-grow: 0; flex-shrink: 0; max-width: 41.66666666666667%; margin-left: 8.333333333333334%;"
          >
            <div
              class="HomePage__Image"
            >
              <img
                alt="Sleigh"
                src="sleigh.png"
                width="80%"
              />
            </div>
          </div>
        </div>
        <span
          style="display: table; clear: both;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
